homeassistant/components/skybell/camera.py

```diff
@@ -70,3 +78,26 @@ async def async_camera_image(
     ) -> bytes | None:
         """Get the latest camera image."""
         return self._device.images[self.entity_description.key]
+
+
+class SkybellActivityCamera(SkybellCamera):
+    """A camera implementation for latest Skybell activity."""
+
+    async def handle_async_mjpeg_stream(
+        self, request: web.Request
+    ) -> web.StreamResponse | None:
+        """Generate an HTTP MJPEG stream from the latest recorded activity."""
+        stream = CameraMjpeg(get_ffmpeg_manager(self.hass).binary)
+        url = await self.coordinator.device.async_get_activity_video_url()
+        await stream.open_camera(url, extra_cmd="-r 210")
+
+        try:
+            return await async_aiohttp_proxy_stream(
+                self.hass,
+                request,
+                await stream.get_reader(),
+                get_ffmpeg_manager(self.hass).ffmpeg_stream_content_type,
+            )
+        except Exception as ex:  # pylint: disable=broad-except
+            LOGGER.error("Error occurred while playing stream: %s", ex)
+            return None
```

Comment:
Usually we let the exception raise here so they know why it failed.  We also need to close the stream

```suggestion
        finally:
            await stream.close()
```

Reply:
I've tried that but it ends up with Broken Pipe if the latest recording was from a demand event. That is when the user uses the Skybell app to open a call. Guess we'll just catch and pass that error?

Reply:
If we are stuck with an exception handler please avoid the broad one

Reply:
As with many issues, this one seems to be intermittent. Tested with new demand recording, works correctly now.